lib/model/model.go

```diff
@@ -1412,18 +1418,16 @@ func (m *model) ccHandleFolders(folders []protocol.Folder, deviceCfg config.Devi
 	}
 
 	indexSenders.removeAllExcept(seenFolders)
-	// All current pending folders were touched above, so discard any with older timestamps
-	expiredPending, err := m.db.RemovePendingFoldersBeforeTime(deviceID, handleTime)
-	if err != nil {
-		l.Infof("Could not clean up pending folder entries: %v", err)
+	for folder := range expiredPending {
+		m.db.RemovePendingFolderForDevice(folder, deviceID)
 	}
 	if len(updatedPending) > 0 || len(expiredPending) > 0 {
-		expiredPendingList := make([]map[string]string, len(expiredPending))
-		for i, folderID := range expiredPending {
-			expiredPendingList[i] = map[string]string{
+		expiredPendingList := make([]map[string]string, 0, len(expiredPending))
```

Comment:
Out of curiosity, what's the benefit of setting the length to zero, then `append()`ing?  Just readability?

Reply:
`make([]Foo, 10)` allocates a slice with pre-allocated 10 items, so appending would result in the 11th item (first 10 having zero'ed values)

`0, len(foo)` allocates 0 sized slice with len(foo) capacity, so we are sure append will not need to re-allocate and copy the slice as we're appending (because it runs out of capacity)

Reply:
I understand that difference, but the previous approach was `L := make(..., 10)` and then *update* items `L[i]`, skipping `append()` completely.  Same overall effect, but any other disadvantages?

Reply:
I don't think so, I prefer the latter as you don't need a variable for the index.